content/blog/2016/2016-05-18-external-workspace-manager-plugin.adoc

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title: "External Workspace Manager Plugin"
+tags:
+- jenkins
+- pipeline
+- plugins
+author: alexsomai
+---
+
+Currently, Jenkins’ build workspace may become very large in size due to the fact that some compilers
+generate very large volumes of data. The existing plugins that share the workspace across builds are able
+to do this by copying the files from one workspace to another, process which is inefficient.
+A solution is to have a Jenkins plugin that is able to manage and reuse the same workspace between multiple
+builds.
+
+As part of the link:https://summerofcode.withgoogle.com/[Google Summer of Code] 2016 I will be working on
+the External Workspace Manager plugin. This plugin aims to provide an external workspace management system.
+It should facilitate workspace share and reuse across multiple Jenkins jobs. It should eliminate the need
+to copy, archive or move files. The plugin will be written for
+link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Plugin[Pipeline] jobs.
+
+The complete project proposal, along with the design details, features, estimated timeline and some implementation
+ideas may be found in this link:https://goo.gl/fq3RAe[Google Doc].
+The plugin repository will be public on link:https://github.com/jenkinsci/external-workspace-manager-plugin[GitHub].
+
+A prototype version of the plugin should be available in late June and the releasable version in late August.
+I will be presenting demos within the community with the plugin functionality.
+
+If you are interested in this plugin, subscribe to the
+link:https://groups.google.com/forum/#!forum/jenkinsci-dev[Jenkins developers] mailing list where I will be
+posting updates from time to time.
```

Comment:
Nice to have something like "I will appreciate any feedback. Feel free to comment in the design document. If you are interested to have a verbal conversation, feel free to join our regular meetings on Mondays (TODO: reference to the time, nice to have a link to something like timeanddate.com with automatic TZ conversion)".